File: snippets/multi-step-agent/snippet-02.py
from glean.api_client import Glean
from glean.api_client.models import ContentType, Message, MessageTextBlock

glean = Glean(api_token=..., instance=...)

response = glean.client.agents.run(
    agent_id=agent_id,
    messages=[Message(role="USER", content=[MessageTextBlock(text=question, type=ContentType.TEXT)])],
    http_headers={"X-Glean-Act-As": "marcus.webb@acme.example.com"},
)

print(response.run.status)
for message in response.messages or []:
    print(message.role, "".join(b.text for b in message.content or []))

File: snippets/permissions-aware-rag/snippet-02.ts
const glean = new Glean({ apiToken: ..., instance: ... });

const response = await glean.search.query({ query: question, page_size: 8 });

const sources = (response.results ?? [])
  .filter((r) => r.title && r.snippets)
  .map((r) => ({
    title: r.title!,
    url: r.url,
    text: r.snippets!.filter(Boolean).join('\n'),
  }));
